tklock.c

```diff
@@ -347,8 +347,7 @@ static gboolean ts_event_control(gboolean enable)
 	execute_datapipe(&touchscreen_suspend_pipe, GINT_TO_POINTER(!enable),
 			USE_INDATA, CACHE_INDATA);
 
-	return generic_event_control(mce_touchscreen_sysfs_disable_path,
```

Comment:
Please explain in the commit message why is this needed.

Reply:
in the old code that function allways returns here: 
https://github.com/maemo-leste/mce/blob/3cbee89d364eea0211fa7a37493e39a6c2b22192/tklock.c#L434

unless the generic_x11_event_control returns false, witch it should never do since that indicates a failure communicating with X11.

because of this generic_event_control never gets called. To preserve the same behavior i had to remove it when removing https://github.com/maemo-leste/mce/blob/3cbee89d364eea0211fa7a37493e39a6c2b22192/tklock.c#L433 (see above as to why that was removed)

Honestly this looks like a possible bug to me, but i dont know and its unrelated to what i was doing anyways so i left it alone.